src/com/spikes2212/genericsubsystem/solenoid/PistonSubsystem.java

```diff
@@ -0,0 +1,64 @@
+package com.spikes2212.genericsubsystem.solenoid;
+
+import java.util.function.Supplier;
+
+import com.spikes2212.utils.Piston;
+
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+public class PistonSubsystem extends Subsystem {
+	protected Piston piston;
+	
+	public PistonSubsystem(String name, Piston piston) {
+		super(name);
+		this.piston = piston;
+
+	}
+
+	public PistonSubsystem(Piston piston) {
+		this.piston = piston;
+	}
+	
+	public int getSpeed() {
+		if(piston.get() == Value.kForward)
+			return 1;
+		if(piston.get() == Value.kReverse)
+			return -1;
+		
+		return 0;
+	}
+	
+	public void stop() {
+		piston.off();
+	}
+
+	public boolean canMove(int speed) {
+		if (speed > 0 && piston.get() != Value.kForward || speed < 0 && piston.get() != Value.kReverse || speed == 0) {
+			return true;
+		}
+		return false;
+	}
+
+	public void move(int speed) {
+		if (canMove(speed)) {
+			if (speed == 0) {
+				piston.off();
+			} else if (speed < 0) {
+				piston.close();
+			} else if (speed > 0) {
+				piston.open();
+			}
+		}
+	}
+

```

Comment:
Why the hell would you do this way. All you need to do is create an open and a close command